parquet-hadoop/src/main/java/org/apache/parquet/hadoop/rewrite/ParquetRewriter.java

```diff
@@ -176,22 +192,40 @@ public ParquetRewriter(RewriteOptions options) throws IOException {
     writer.start();
   }
 
+  private MessageType getSchema() {
+    MessageType schemaMain = inputFiles.peek().getFooter().getFileMetaData().getSchema();
+    if (inputFilesToJoin.isEmpty()) {
+      return schemaMain;
+    } else {
+      Map<String, Type> fieldNames = new LinkedHashMap<>();
+      schemaMain.getFields().forEach(x -> fieldNames.put(x.getName(), x));
+      inputFilesToJoin
+          .peek()
+          .getFooter()
+          .getFileMetaData()
+          .getSchema()
+          .getFields()
+          .forEach(x -> {
+            if (!fieldNames.containsKey(x.getName()) || overwriteInputWithJoinColumns) {
+              fieldNames.put(x.getName(), x);
```

Comment:
Please add a log to indicate `overwriteInputWithJoinColumns` takes effect for any column

Reply:
done